### PR TITLE
refactor: isSameAddress

### PR DESCRIPTION
**Pull request by Himself65**
closes: #3409

### CODE REVIEW

File: packages/maskbook/src/extension/background-script/EthereumServices/wallet.ts [hunk 1]
```diff
@@ -17,7 +17,8 @@ revalidateWallets()
 //#endregion
 
 export function getWalletCached(address = currentAccountSettings.value) {
-    return wallets.find((x) => isSameAddress(x.address, address))
+    const matchAddress = MatchAddress(address)
+    return wallets.find((x) => matchAddress(x.address))
```
Comment (septs) [inline packages/maskbook/src/extension/background-script/EthereumServices/wallet.ts:21]:
```suggestion
    const matchAddress = 
    return wallets.find(MatchAddress(address))
```

File: packages/web3-shared/src/utils/index.ts [hunk 2]
```diff
@@ -12,9 +12,9 @@ import {
     Web3Constants,
 } from '../types'
 
-export function isSameAddress(addrA: string, addrB: string) {
-    return addrA.toLowerCase() === addrB.toLowerCase()
-}
+export const isSameAddress = (addrA: string, addrB: string) => addrA.toLowerCase() === addrB.toLowerCase()
+
+export const MatchAddress = (a: string) => (b: string) => isSameAddress(a, b)
```
Comment (Jack-Works) [inline packages/web3-shared/src/utils/index.ts:17]: Why it's uppercased?

File: packages/maskbook/src/extension/options-page/DashboardComponents/WalletAssetsTable.tsx [hunk 3]
```diff
@@ -139,9 +140,7 @@ function ViewDetailed(props: ViewDetailedProps) {
                     }}>
                     <Typography className={classes.name} color="textPrimary" component="span">
                         {new BigNumber(formatBalance(asset.balance, asset.token.decimals)).toFixed(
-                            stableTokens.some((y: ERC20TokenDetailed) => isSameAddress(y.address, asset.token.address))
-                                ? 2
-                                : 6,
+                            stableTokens.some((y: ERC20TokenDetailed) => matchTokenAddress(y.address)) ? 2 : 6,
```
Comment (septs) [inline packages/maskbook/src/extension/options-page/DashboardComponents/WalletAssetsTable.tsx:143]: think ...., need 2nd parameter

```
function MatchAddress(input: string)
function MatchAddress<T extends object>(input: T, field: string)
```

File: packages/maskbook/src/extension/background-script/EthereumServices/wallet.ts [hunk 4]
```diff
@@ -17,7 +17,7 @@ revalidateWallets()
 //#endregion
 
 export function getWalletCached(address = currentAccountSettings.value) {
-    return wallets.find((x) => isSameAddress(x.address, address))
+    return wallets.find(MatchAddress(address))
```
Comment (septs) [inline packages/maskbook/src/extension/background-script/EthereumServices/wallet.ts:20]: sorry, revert

File: packages/web3-shared/src/utils/index.ts (hunk 2 above)
Comment (septs) [reply packages/web3-shared/src/utils/index.ts:17]: my idea......

Comment (Jack-Works) [reply packages/web3-shared/src/utils/index.ts:17]: Can you explain why?

Comment (septs) [reply packages/web3-shared/src/utils/index.ts:17]: https://github.com/DimensionDev/Maskbook/issues/3409

Comment (Jack-Works) [reply packages/web3-shared/src/utils/index.ts:17]: I mean why it should be `Match` not `match`

**Comment by Jack-Works** [comment]
Linters and tests / type-check (pull_request) Failing after 2m — type-check

**Comment by guanbinrui** [comment]
I've renamed dev to develop. Please reopen it against the `develop` branch later.

**Pull request closed without merging**